pkg/frontend/querymiddleware/labels_query_cache.go

```diff
@@ -124,8 +121,12 @@ func generateLabelsQueryRequestCacheKey(startTime, endTime int64, labelName stri
 	return b.String()
 }
 
-func parseRequestTimeParam(req *http.Request, paramName string, defaultValue int64) (int64, error) {
-	value := req.FormValue(paramName)
+func parseRequestTimeParam(values url.Values, paramName string, defaultValue int64) (int64, error) {
+	var value string
+	if len(values[paramName]) > 0 {
```

Comment:
Note to reviewers: this is the same thing done by `req.FormValue()` (it picks the 1st one).